pkg/querier/blocks_store_queryable.go

```diff
@@ -292,14 +294,74 @@ func (q *blocksStoreQuerier) Select(_ bool, sp *storage.SelectHints, matchers ..
 	return q.selectSorted(sp, matchers...)
 }
 
-func (q *blocksStoreQuerier) LabelValues(name string) ([]string, storage.Warnings, error) {
-	// Cortex doesn't use this. It will ask ingesters for metadata.
-	return nil, nil, errors.New("not implemented")
+func (q *blocksStoreQuerier) LabelNames() ([]string, storage.Warnings, error) {
```

Comment:
I would like to see unit tests for this. You could get inspired by `Select()` unit tests.